builtin/game/chatcommands.lua

```diff
@@ -530,11 +530,16 @@ core.register_chatcommand("giveme", {
 })
 
 core.register_chatcommand("spawnentity", {
-	params = "<EntityName>",
-	description = "Spawn entity at your position",
+	params = "<EntityName> [<X>,<Y>,<Z>]",
+	description = "Spawn entity at given (or your) position",
 	privs = {give=true, interact=true},
 	func = function(name, param)
-		local entityname = string.match(param, "(.+)$")
+		local entityname = string.match(param, "^([^ ]+)$")
+		local p
+		if not entityname then
```

Comment:
there should really be one standard way of passing the pos.
and the teleport function look like a horrible mess,  no wonder people keep saying the codebase is horribly coded
in any case, spliting the entityname from pos before passing it to the code which change it to a table would be nicer

btw, this doesn't even work, you forgot the tonumber, so you are passing a string instead of a number
ERROR[main]: ServerError: Invalid position coordinate 'x' (expected number got string).